libs/@guardian/libs/src/storage/README.md

```diff
@@ -119,6 +119,31 @@ Removes all items from storage.
 storage.local.clear();
 ```
 
+## `keys()`
+
+Returns: `string[] | undefined`
+
+Get all of the keys in storage.
+
+### Notes
+
+This is not a method in the native API, and gets all keys by iterating over the
+entire storage object. It is therefore potentially subject to race conditions
```

Comment:
> It is therefore potentially subject to race conditions

this makes me wonder if it's something we should have. is it ok to offer a method that might return false information? how is a user supposed to know that? especially if your reading an existing usage that someone else (who did know) wrote?

is it better to some how force ourselves to confront that at the point of doing it, rather than wrapping it in something that _appears_ trustworthy (but can't be)?

Reply:
I'll investigate a little more to find out how realistic this issue is.

If we are concerned about this then that would speak in favour of replacing this method with a simple wrapper around the built-in `key` method, as discussed above 👍

Reply:
(partial update here: https://github.com/guardian/csnx/pull/692#discussion_r1261365972)